dlib/python/numpy_image.h

```diff
@@ -70,11 +73,9 @@ class numpy_rgb_image
     numpy_rgb_image (boost::python::object& img) 
     {
         long shape[3];
-        get_numpy_ndarray_parts(img, _data, shape);
+        get_numpy_ndarray_parts(img, _data, _contig_buf, shape);
         _nr = shape[0];
         _nc = shape[1];
-        if (shape[2] != 3)
```

Comment:
Oh, I see.  In that case, the if statements in get_numpy_ndarray_parts() should be adjusted to make sure dims==3 for this case.  Right now they might give the wrong result for a grayscale image that just happened to have 3 columns.